src/library/dao/playlistdao.cpp

```diff
@@ -980,18 +982,16 @@ void PlaylistDAO::shuffleTracks(const int playlistId, const QList<int>& position
         trackPositionIds.insert(trackBPosition, trackAId);
         newPositions.swap(newPositions.indexOf(trackAPosition),
                           newPositions.indexOf(trackBPosition));
-        QString swapQuery = "UPDATE PlaylistTracks SET position=%1 "
-                "WHERE position=%2 AND playlist_id=%3";
-        query.exec(swapQuery.arg(QString::number(-1),
-                                 QString::number(trackAPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackAPosition),
-                                 QString::number(trackBPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackBPosition),
-                                 QString::number(-1),
-                                 QString::number(playlistId)));
 
+        qSwap(playListTrackIds[trackAPosition - 1],
+              playListTrackIds[trackBPosition - 1]);
```

Comment:
Yes, I tried to think of it, I but on this I got the problem, sometimes the trackAPosition and trackBPosition gives the duplicate values, i.e. values repeat, so if we do changes in the database  in the loop from where we get the trackAPositions and trackBPositions it creates error, and if we get the new positions in some new list newPositions, then we do shuffling in other loop it doesn't make any difference like the thing which we achieve now will same as we will achieve in that way, except that by using Hash tables we get the index starting from 1 instead of 0, 
Yes, I tried to think of it, I but on this I got the problem, sometimes the trackAPosition and trackBPosition gives the duplicate values, i.e. values repeat, so if we do changes in the database  in the loop from where we get the trackAPositions and trackBPositions it creates error, and if we get the new positions in some new list newPositions, then we do shuffling in other loop it doesn't make any difference like the thing which we achieve now will same as we will achieve in that way, except that by using Hash tables we get the index starting from 1 instead of 0,